plugin-server/src/utils/db/db.ts

```diff
@@ -561,6 +598,100 @@ export class DB {
+    //         return null
+    //     }
+    //     const props = await this.redisGet(this.getPersonIdCacheKey(teamId, distinctId), null)
+    //     if (personId) {
+    //         this.statsd?.increment(`person_info_cache.hit`, { lookup: 'person_id', team_id: teamId.toString() })
+    //         return Number(personId)
+    //     }
+    //     this.statsd?.increment(`person_info_cache.miss`, { lookup: 'person_id', team_id: teamId.toString() })
+    //     // Query from postgres and update cache
+    //     const result = await this.postgresQuery(
+    //         'SELECT person_id FROM posthog_persondistinctid WHERE team_id=$1 AND distinct_id=$2 LIMIT 1',
+    //         [teamId, distinctId],
+    //         'fetchPersonId'
+    //     )
+    //     if (result.rows.length > 0) {
+    //         const personId = Number(result.rows[0].person_id)
+    //         await this.updatePersonIdCache(teamId, distinctId, personId)
+    //         return personId
+    //     }
+    //     return null
+    // }
+
+    private async getGroupProperty(teamId: number, groupIdentifier: GroupIdentifier): Promise<GroupProperties> {
+        const props = await this.redisGet(
+            this.getGroupPropertiesCacheKey(teamId, groupIdentifier.index, groupIdentifier.key),
+            null
+        )
+        return { identifier: groupIdentifier, properties: props !== null ? (props as Properties) : null }
+    }
+
+    // TODO: add test for this
+    private async getGroupPropertiesFromDbAndUpdateCache(
+        teamId: number,
+        groupIdentifiers: GroupIdentifier[]
+    ): Promise<Record<string, string>> {
+        const query_options: string[] = []
+        const args: any[] = [teamId]
+        let index = 2
+        for (const gi of groupIdentifiers) {
+            query_options.push(`(group_type_index = ${index} group_key = ${index + 1} AND)`)
+            index += 2
+            args.push([gi.index, gi.key])
+        }
+        const result = await this.postgresQuery(
+            'SELECT group_type_index, group_key, group_properties FROM posthog_group WHERE team_id=$1 AND '.concat(
+                query_options.join(' OR ')
+            ),
+            args,
+            'fetchGroupProperties'
+        )
+        const promises = []
+        const res: Record<string, string> = {}
+        for (const row in result.rows) {
+            // TODO: why do you think its a string???
+            const index = Number(row.group_type_index)
+            const key = String(row.group_key)
+            const properties = row.group_properties as Properties
+            promises.push(this.updateGroupPropertiesCache(teamId, index, key, properties))
+            res[`group${index}_properties`] = JSON.stringify(properties)
+        }
+        await Promise.all(promises)
+        return res
+    }
+
+    public async getGroupProperties(teamId: number, groups: GroupIdentifier[]): Promise<Record<string, string>> {
```

Comment:
I wrote a suggestion elsewhere that will fix this but currently this function will accept `undefined` values for identifiers and doesn't handle it correctly

Reply:
fixed the call location